### ISSUE TITLE
TypedDict `values()` and `items()` infer incorrect `object` type for dict value.

### ISSUE THREAD

**Issue by relsunkaev** [microsoft/pyright#3129]
**Describe the bug**
When using `values()` and `items()` member functions of `TypedDict`, pyright infers the "value" element to be of type `object`.

**To Reproduce**
```python
from typing import TypedDict
class TestDict(TypedDict):
    a: list[int]
    b: list[int]
test_dict = TestDict(a=[], b=[])
for value in test_dict.values():
    value.append(0)
```
produces the following when type checked
```bash
/.../test.py:7:4 - error: Type of "append" is unknown (reportUnknownMemberType)
```

**Comment by erictraut** [comment]
This is expected behavior. The `values` method for `TypedDict` comes from the `_TypedDict` class declaration in `typing.pyi`:

```python
class _TypedDict(Mapping[str, object], metaclass=ABCMeta):
    ...
    def values(self) -> ValuesView[object]: ...
    ...
```

TypedDicts generally have various types for their values, so `ValuesView[object]` is a reasonable type for `values()`.

If you want to benefit from the type checker's knowledge of individual fields within a TypedDict, you should reference those fields individually, not through the `values()` method.

```python
test_dict["a"].append(0)
test_dict["b"].append(0)
```
